### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by brettimus**
This PR adds some friendlier context and instructions to the README file.

In a follow-up, we should link to the honc.dev website, as well as any additional documentation we have on the template itself.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,45 @@
+# Honc Honc!
+
+## Quickstart
+
+This is a template of a Hono-Neon-Drizzle-Cloudflare project.
```
Comment (mies) [inline README.md:5]: "For a Hono-Neon...."??

File: README.md [hunk 2]
```diff
@@ -1,21 +1,45 @@
+# Honc Honc!
+
+## Quickstart
+
+This is a template of a Hono-Neon-Drizzle-Cloudflare project. 
+
+Use this template as follows:
+
+```sh
+npm create cloudflare@latest -- --template=git@github.com:fiberplane/honc-template.git
```
Comment (mies) [inline README.md:10]: Add perhaps pnpm and yarn options?

File: README.md [hunk 3]
```diff
@@ -1,21 +1,45 @@
+# Honc Honc!
```
Comment (mies) [inline README.md:1]: Add the goose logo 🪿

File: README.md (hunk 2 above)
Comment (brettimus) [reply README.md:18]: added a comment!

File: README.md (hunk 1 above)
Comment (brettimus) [reply README.md]: explained each component in a list

**Review by mies — APPROVED** [review]

**Pull request merged**